superset/migrations/versions/d416d0d715cc_add_limiting_factor_column_to_query_.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""add_limiting_factor_column_to_query_model.py
+
+Revision ID: d416d0d715cc
+Revises: 19e978e1b9c3
+Create Date: 2021-04-16 17:38:40.342260
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "d416d0d715cc"
+down_revision = "19e978e1b9c3"
+
+from enum import Enum
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.dialects.postgresql.base import PGDialect
+
+limiting_factor = postgresql.ENUM(
+    "DROPDOWN",
+    "QUERY",
+    "NOT_LIMITED",
+    "QUERY_AND_DROPDOWN",
+    "UNKNOWN",
+    name="limitingfactor",
+)
+
+
+def upgrade():
+    bind = op.get_bind()
+    if isinstance(bind.dialect, PGDialect):
+        limiting_factor.create(bind)
+    with op.batch_alter_table("query") as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "limiting_factor",
+                sa.Enum(
```

Comment:
We tend to prefer not using Enums, because they force a new DB migration when a new value is needed, You can still use python `Enum` but use a String on the DB field

Reply:
hey! I saw the comments in the other channel, I just had my second shot two days ago, and was extremely under the weather yesterday. What do you think of Craig's suggestion to use a varchar instead?